airflow/executors/base_executor.py

```diff
@@ -141,13 +141,27 @@ def fail(self, key):
     def success(self, key):
         self.change_state(key, State.SUCCESS)
 
-    def get_event_buffer(self):
+    def get_event_buffer(self, dag_ids=None):
```

Comment:
Can I suggest a slight refactor to remove some duplication and use the dict.pop return value:

```python
cleared_events = {}
if dag_ids is None:
    cleared_events = self.event_buffer
    self.event_buffer = {}
else:
    for key in list(previous_buffer):
        dag_id, _, _ = key
        if dag_id in dag_ids:
            cleared_events[key] = self.event_buffer.pop(key)
return cleared_events
```

Reply:
I will, will slightly change yours